packages/design-system/src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,123 @@
+import FocusTrap from 'focus-trap-react';
+import Button from '../Button/Button';
+import React, { useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import uniqueId from 'lodash/uniqueId';
+
+// TODO: closeButtonText, heading should be a string, but it is being used as a node in MCT,
+// until we provide a better solution for customization, we type it as a node.
+export interface DrawerProps {
+  /**
+   * Gives more context to screen readers on the Drawer close button.
+   */
+  ariaLabel?: string;
+  closeButtonText?: React.ReactNode;
+  children: React.ReactNode;
+  className?: string;
+  footerBody: React.ReactNode;
+  footerTitle: string;
+  /**
+   * Enables focus trap functionality within Drawer.
+   */
+  hasFocusTrap?: boolean;
+  /**
+   * Text for the Drawer heading. Required because the `heading` will be focused on mount.
+   */
+  heading: string | React.ReactNode;
+  /**
+   * A unique `id` to be used on heading element to label multiple instances of Drawer.
+   */
+  headingId?: string;
+  /**
+   * Heading type to override default `<h3>`
+   */
+  headingLevel: '1' | '2' | '3' | '4' | '5';
+  /**
+   * Enables "sticky" position of Drawer header element.
+   */
+  isHeaderSticky?: boolean;
+  /**
+   * Enables "sticky" position of Drawer footer element.
+   */
+  isFooterSticky?: boolean;
+  onCloseClick: (event: React.MouseEvent | React.KeyboardEvent) => void;
+}
+
+const Drawer = (props: DrawerProps) => {
+  const headingRef = useRef(null);
+  const id: string = props.headingId || uniqueId('drawer_');
```

Comment:
In order to avoid generating a different unique id on each render, you can use `useRef` here as well like https://github.com/CMSgov/design-system/blob/typescript-conversions/packages/design-system/src/components/Tooltip/Tooltip.tsx#L86